src/client_side.cc

```diff
@@ -2547,69 +2545,89 @@ httpsCreate(const Comm::ConnectionPointer &conn, const Security::ContextPointer
 
 /**
  *
- * \retval 1 on success
- * \retval 0 when needs more data
- * \retval -1 on error
+ * \retval true on success
+ * \retval false when needs more data
+ * \retval false when an error occured and closes the TCP connection
  */
-static int
-Squid_SSL_accept(ConnStateData *conn, PF *callback)
+static bool
+tryTlsHandshake(ConnStateData *conn, PF *callback)
```

Comment:
This function is where the "TLS connection" concept using handshakes as a pseudo-accept() starts to fall down. It is the callback run after TCP accept(). It *attempts* to perform a TLS handshake, but may result in 0-N handshakes happening - in that 0 case there is "anonymous TLS", aka TLS with no session / "connection".

It just happens that OpenSSL API is not capable of the latter TLS/1.2+ feature so it uses the badly named function "SSL_accept()" and dies if no handshake occurs.

I'm using 'attempt' instead of 'try', but not adding 'accept' into the naming.

Reply:
> This function is where the "TLS connection" concept using handshakes as a pseudo-accept() starts to fall down

I see no evidence of that. Yes, accepts may happen at different layers (e.g., TCP, TLS, and application). There is nothing wrong or even confusing about that IMO.

> It attempts to perform a TLS handshake, but may result in 0-N handshakes happening

Counting the number of handshakes sounds like splitting hairs to me. I doubt the concept of a handshake (if it is formally defined) allows for such counting and even if it is, there seems to be no point in counting individual "shakes" when we are talking about naming a function that does them.

> that 0 case there is "anonymous TLS", aka TLS with no session / "connection".

My quick search for these terms failed to find any relevant matches. If you are sure this is something we should be discussing, please provide pointers defining the relevant "anonymous TLS" concept.

> I'm using 'attempt' instead of 'try', but not adding 'accept' into the naming.

I am considering this problem as unresolved. I am not married to the word "accept" (even though the term fits the bill very well), but we should (continue to) distinguish client-Squid handshakes from Squid-server handshakes.